Theia does not run with Node.js `v10.3.0`

Help wanted. See: https://github.com/theia-ide/theia/pull/3358#issuecomment-434978417
```
$ nvm use 10 && node --version && npm --version && yarn --version && python --version && git clone https://github.com/theia-ide/theia.git && cd theia && yarn && cd examples/browser && yarn run start
Now using node v10.3.0 (npm v6.1.0)
v10.3.0
6.1.0
1.5.1
Python 2.7.10
Cloning into 'theia'...
remote: Counting objects: 25608, done.
remote: Compressing objects: 100% (94/94), done.
remote: Total 25608 (delta 41), reused 42 (delta 18), pack-reused 25495
Receiving objects: 100% (25608/25608), 118.12 MiB | 2.32 MiB/s, done.
Resolving deltas: 100% (17318/17318), done.
yarn install v1.5.1
(node:5029) [DEP0005] DeprecationWarning: Buffer() is deprecated due to security and usability issues. Please use the Buffer.alloc(), Buffer.allocUnsafe(), or Buffer.from() methods instead.
[1/5] 🔍  Validating package.json...
[2/5] 🔍  Resolving packages...
[3/5] 🚚  Fetching packages...
warning vscode-base-languageclient@0.0.1-alpha.5: The engine "vscode" appears to be invalid.
[4/5] 🔗  Linking dependencies...
warning " > istanbul-instrumenter-loader@3.0.1" has unmet peer dependency "webpack@^2.0.0 || ^3.0.0 || ^4.0.0".
warning "workspace-aggregator-b404bb6e-1773-4313-be3f-6c77f10d80b0 > @theia/application-manager > font-awesome-webpack@0.0.5-beta.2" has unmet peer dependency "font-awesome@>=4.3.0".
[5/5] 📃  Building fresh packages...
[1/11] ⠁ fibers:                               ^
[2/11] ⠁ phantomjs-prebuilt: Receiving...
[3/11] ⠁ dtrace-provider: ACTION binding_gyp_ndtp_target_build_ndtp .
[7/11] ⠁ node-pty: CXX(target) Release/obj.target/pty/src/unix/pty.o
error An unexpected error occurred: "/Users/akos.kitta/Desktop/theia/node_modules/fibers: Command failed.
Exit code: 127
Command: sh
Arguments: -c node build.js || nodejs build.js
Directory: /Users/akos.kitta/Desktop/theia/node_modules/fibers
Output:
gyp info it worked if it ends with ok
gyp info using node-gyp@3.6.2
gyp info using node@10.3.0 | darwin | x64
gyp http GET https://nodejs.org/download/release/v10.3.0/node-v10.3.0-headers.tar.gz
gyp http 200 https://nodejs.org/download/release/v10.3.0/node-v10.3.0-headers.tar.gz
gyp http GET https://nodejs.org/download/release/v10.3.0/SHASUMS256.txt
gyp http 200 https://nodejs.org/download/release/v10.3.0/SHASUMS256.txt
gyp info spawn /usr/bin/python
gyp info spawn args [ '/Users/akos.kitta/Desktop/theia/node_modules/node-gyp/gyp/gyp_main.py',
gyp info spawn args   'binding.gyp',
gyp info spawn args   '-f',
gyp info spawn args   'make',
gyp info spawn args   '-I',
gyp info spawn args   '/Users/akos.kitta/Desktop/theia/node_modules/fibers/build/config.gypi',
gyp info spawn args   '-I',
gyp info spawn args   '/Users/akos.kitta/Desktop/theia/node_modules/node-gyp/addon.gypi',
gyp info spawn args   '-I',
gyp info spawn args   '/Users/akos.kitta/.node-gyp/10.3.0/include/node/common.gypi',
gyp info spawn args   '-Dlibrary=shared_library',
gyp info spawn args   '-Dvisibility=default',
gyp info spawn args   '-Dnode_root_dir=/Users/akos.kitta/.node-gyp/10.3.0',
gyp info spawn args   '-Dnode_gyp_dir=/Users/akos.kitta/Desktop/theia/node_modules/node-gyp',
gyp info spawn args   '-Dnode_lib_file=/Users/akos.kitta/.node-gyp/10.3.0/<(target_arch)/node.lib',
gyp info spawn args   '-Dmodule_root_dir=/Users/akos.kitta/Desktop/theia/node_modules/fibers',
gyp info spawn args   '-Dnode_engine=v8',
gyp info spawn args   '--depth=.',
gyp info spawn args   '--no-parallel',
gyp info spawn args   '--generator-output',
gyp info spawn args   'build',
gyp info spawn args   '-Goutput_dir=.' ]
xcode-select: error: tool 'xcodebuild' requires Xcode, but active developer directory '/Library/Developer/CommandLineTools' is a command line tools instance

xcode-select: error: tool 'xcodebuild' requires Xcode, but active developer directory '/Library/Developer/CommandLineTools' is a command line tools instance

gyp info spawn make
gyp info spawn args [ 'BUILDTYPE=Release', '-C', 'build' ]
  CXX(target) Release/obj.target/fibers/src/fibers.o
../src/fibers.cc:122:64: error: too few arguments to function call, expected at least 3, have 2
                return String::NewFromOneByte(isolate, (const uint8_t*)string);
                       ~~~~~~~~~~~~~~~~~~~~~~                                ^
/Users/akos.kitta/.node-gyp/10.3.0/include/node/v8.h:2736:3: note: 'NewFromOneByte' declared here
  static V8_WARN_UNUSED_RESULT MaybeLocal<String> NewFromOneByte(
  ^
../src/fibers.cc:126:64: error: too few arguments to function call, expected at least 3, have 2
                return String::NewFromOneByte(isolate, (const uint8_t*)string);
                       ~~~~~~~~~~~~~~~~~~~~~~                                ^
/Users/akos.kitta/.node-gyp/10.3.0/include/node/v8.h:2736:3: note: 'NewFromOneByte' declared here
  static V8_WARN_UNUSED_RESULT MaybeLocal<String> NewFromOneByte(
  ^
../src/fibers.cc:406:24: warning: 'Utf8Value' is deprecated [-Wdeprecated-declarations]
                                        String::Utf8Value stack(uni::Deref(that.isolate, fatal_stack));
                                                          ^
/Users/akos.kitta/.node-gyp/10.3.0/include/node/v8.h:2821:5: note: 'Utf8Value' has been explicitly marked deprecated here
    V8_DEPRECATED(\"Use Isolate version\",
    ^
/Users/akos.kitta/.node-gyp/10.3.0/include/node/v8config.h:321:29: note: expanded from macro 'V8_DEPRECATED'
  declarator __attribute__((deprecated))
                            ^
../src/fibers.cc:433:80: error: no matching member function for call to 'NewInstance'
                                return uni::Return(uni::Deref(Isolate::GetCurrent(), tmpl)->GetFunction()->NewInstance(1, argv), args);
                                                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~
/Users/akos.kitta/.node-gyp/10.3.0/include/node/v8.h:3862:44: note: candidate function not viable: requires single argument 'context', but 2 arguments were provided
  V8_WARN_UNUSED_RESULT MaybeLocal<Object> NewInstance(
                                           ^
/Users/akos.kitta/.node-gyp/10.3.0/include/node/v8.h:3859:44: note: candidate function not viable: requires 3 arguments, but 2 were provided
  V8_WARN_UNUSED_RESULT MaybeLocal<Object> NewInstance(
                                           ^
../src/fibers.cc:621:14: error: no matching constructor for initialization of 'v8::TryCatch'
                                TryCatch try_catch;
                                         ^
/Users/akos.kitta/.node-gyp/10.3.0/include/node/v8.h:8282:12: note: candidate constructor not viable: requires single argument 'isolate', but no arguments were provided
  explicit TryCatch(Isolate* isolate);
           ^
/Users/akos.kitta/.node-gyp/10.3.0/include/node/v8.h:8402:3: note: candidate constructor not viable: requires 1 argument, but 0 were provided
  TryCatch(const TryCatch&) = delete;
  ^
../src/fibers.cc:628:13: warning: 'Compile' is deprecated [-Wdeprecated-declarations]
                                Script::Compile(uni::NewLatin1String(that.isolate, \"void 0;\"));
                                        ^
/Users/akos.kitta/.node-gyp/10.3.0/include/node/v8.h:1234:10: note: 'Compile' has been explicitly marked deprecated here
  static V8_DEPRECATED(\"Use maybe version\",
         ^
/Users/akos.kitta/.node-gyp/10.3.0/include/node/v8config.h:321:29: note: expanded from macro 'V8_DEPRECATED'
  declarator __attribute__((deprecated))
                            ^
../src/fibers.cc:741:34: error: no matching member function for call to 'ToNumber'
                        Coroutine::pool_size = value->ToNumber()->Value();
                                               ~~~~~~~^~~~~~~~
/Users/akos.kitta/.node-gyp/10.3.0/include/node/v8.h:2366:44: note: candidate function not viable: requires single argument 'context', but no arguments were provided
  V8_WARN_UNUSED_RESULT MaybeLocal<Number> ToNumber(
                                           ^
/Users/akos.kitta/.node-gyp/10.3.0/include/node/v8.h:2383:35: note: candidate function not viable: requires single argument 'isolate', but no arguments were provided
                    Local<Number> ToNumber(Isolate* isolate) const);
                                  ^
../src/fibers.cc:793:72: error: too few arguments to function call, expected at least 3, have 2
                        fn->SetAccessor(uni::NewLatin1Symbol(isolate, \"current\"), GetCurrent);
                        ~~~~~~~~~~~~~~~                                                     ^
/Users/akos.kitta/.node-gyp/10.3.0/include/node/v8.h:3214:3: note: 'SetAccessor' declared here
  V8_WARN_UNUSED_RESULT Maybe<bool> SetAccessor(
  ^
/Users/akos.kitta/.node-gyp/10.3.0/include/node/v8config.h:410:31: note: expanded from macro 'V8_WARN_UNUSED_RESULT'
#define V8_WARN_UNUSED_RESULT __attribute__((warn_unused_result))
                              ^
../src/fibers.cc:794:63: error: no viable conversion from 'uni::FunctionType (Local<v8::String>, const uni::GetterCallbackInfo &)' (aka 'void (Local<v8::String>, const PropertyCallbackInfo<v8::Value> &)') to 'Local<v8::Name>'
                        fn->SetAccessor(uni::NewLatin1Symbol(isolate, \"poolSize\"), GetPoolSize, SetPoolSize);
                                                                                   ^~~~~~~~~~~
/Users/akos.kitta/.node-gyp/10.3.0/include/node/v8.h:202:7: note: candidate constructor (the implicit copy constructor) not viable: no known conversion from 'uni::FunctionType (Local<v8::String>, const uni::GetterCallbackInfo &)' (aka 'void (Local<v8::String>, const PropertyCallbackInfo<v8::Value> &)') to 'const v8::Local<v8::Name> &' for 1st argument
class Local {
      ^
/Users/akos.kitta/.node-gyp/10.3.0/include/node/v8.h:202:7: note: candidate constructor (the implicit move constructor) not viable: no known conversion from 'uni::FunctionType (Local<v8::String>, const uni::GetterCallbackInfo &)' (aka 'void (Local<v8::String>, const PropertyCallbackInfo<v8::Value> &)') to 'v8::Local<v8::Name> &&' for 1st argument
/Users/akos.kitta/.node-gyp/10.3.0/include/node/v8.h:206:13: note: candidate template ignored: could not match 'Local<type-parameter-0-0>' against 'void (*)(v8::Local<v8::String>, const v8::PropertyCallbackInfo<v8::Value> &)'
  V8_INLINE Local(Local<S> that)
            ^
/Users/akos.kitta/.node-gyp/10.3.0/include/node/v8.h:3215:43: note: passing argument to parameter 'name' here
      Local<Context> context, Local<Name> name,
                                          ^
../src/fibers.cc:795:84: error: too few arguments to function call, expected at least 3, have 2
                        fn->SetAccessor(uni::NewLatin1Symbol(isolate, \"fibersCreated\"), GetFibersCreated);
                        ~~~~~~~~~~~~~~~                                                                 ^
/Users/akos.kitta/.node-gyp/10.3.0/include/node/v8.h:3214:3: note: 'SetAccessor' declared here
  V8_WARN_UNUSED_RESULT Maybe<bool> SetAccessor(
  ^
/Users/akos.kitta/.node-gyp/10.3.0/include/node/v8config.h:410:31: note: expanded from macro 'V8_WARN_UNUSED_RESULT'
#define V8_WARN_UNUSED_RESULT __attribute__((warn_unused_result))
                              ^
In file included from ../src/fibers.cc:1:
In file included from ../src/coroutine.h:1:
In file included from /Users/akos.kitta/.node-gyp/10.3.0/include/node/node.h:63:
/Users/akos.kitta/.node-gyp/10.3.0/include/node/v8.h:213:5: error: assigning to 'v8::Context *' from incompatible type 'v8::String *'
    TYPE_CHECK(T, S);
    ^~~~~~~~~~~~~~~~
/Users/akos.kitta/.node-gyp/10.3.0/include/node/v8.h:167:39: note: expanded from macro 'TYPE_CHECK'
    *(static_cast<T* volatile*>(0)) = static_cast<S*>(0);      \\
                                      ^~~~~~~~~~~~~~~~~~
../src/fibers.cc:794:20: note: in instantiation of function template specialization 'v8::Local<v8::Context>::Local<v8::String>' requested here
                        fn->SetAccessor(uni::NewLatin1Symbol(isolate, \"poolSize\"), GetPoolSize, SetPoolSize);
                                        ^
2 warnings and 9 errors generated.
make: *** [Release/obj.target/fibers/src/fibers.o] Error 1
gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/Users/akos.kitta/Desktop/theia/node_modules/node-gyp/lib/build.js:258:23)
gyp ERR! stack     at ChildProcess.emit (events.js:182:13)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:237:12)
gyp ERR! System Darwin 17.4.0
gyp ERR! command \"/Users/akos.kitta/.nvm/versions/node/v10.3.0/bin/node\" \"/Users/akos.kitta/Desktop/theia/node_modules/.bin/node-gyp\" \"rebuild\" \"--release\"
gyp ERR! cwd /Users/akos.kitta/Desktop/theia/node_modules/fibers
gyp ERR! node -v v10.3.0
gyp ERR! node-gyp -v v3.6.2
gyp ERR! not ok 
node-gyp exited with code: 1
Please make sure you are using a supported platform and node version. If you
would like to compile fibers on this machine please make sure you have setup your
build environment--
Windows + OS X instructions here: https://github.com/nodejs/node-gyp
Ubuntu users please run: `sudo apt-get install g++ build-essential`
```

This seems to be [fixed in version `1.1.0`](https://github.com/Axosoft/nsfw/pull/53#issuecomment-424431122). I have verified it with `nsfw@1.1.1`, and out upstream dependency, `vscode-nsfw`, [already has this change in `1.1.1`](https://github.com/Axosoft/nsfw/commit/52504880f89f99373ae4e141771cdbda4943ee6e).

```
rm -rf nsfw-node-10 \
&& mkdir nsfw-node-10 \
&& cd nsfw-node-10 \
&& nvm install 10 \
&& nvm use 10 \
&& node --version \
&& npm init -y \
&& npm i -S nsfw@1.1.1
```

```
prprpr:Desktop akos.kitta$ rm -rf nsfw-node-10 \
> && mkdir nsfw-node-10 \
> && cd nsfw-node-10 \
> && nvm install 10 \
> && nvm use 10 \
> && node --version \
> && npm init -y \
> && npm i -S nsfw@1.1.1
Downloading https://nodejs.org/dist/v10.13.0/node-v10.13.0-darwin-x64.tar.xz...
######################################################################## 100.0%
Now using node v10.13.0 (npm v6.4.1)
Now using node v10.13.0 (npm v6.4.1)
v10.13.0
Wrote to /Users/akos.kitta/Desktop/nsfw-node-10/package.json:

{
  "name": "nsfw-node-10",
  "version": "1.0.0",
  "description": "",
  "main": "index.js",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "keywords": [],
  "author": "",
  "license": "ISC"
}



> nsfw@1.1.1 install /Users/akos.kitta/Desktop/nsfw-node-10/node_modules/nsfw
> node-gyp rebuild

gyp WARN download NVM_NODEJS_ORG_MIRROR is deprecated and will be removed in node-gyp v4, please use NODEJS_ORG_MIRROR
gyp WARN download NVM_NODEJS_ORG_MIRROR is deprecated and will be removed in node-gyp v4, please use NODEJS_ORG_MIRROR
gyp WARN download NVM_NODEJS_ORG_MIRROR is deprecated and will be removed in node-gyp v4, please use NODEJS_ORG_MIRROR
  CXX(target) Release/obj.target/nsfw/src/NSFW.o
In file included from ../src/NSFW.cpp:1:
In file included from ../src/../includes/NSFW.h:5:
In file included from ../../nan/nan.h:222:
In file included from ../../nan/nan_new.h:189:
../../nan/nan_implementation_12_inl.h:340:28: warning: 'New' is deprecated [-Wdeprecated-declarations]
  return v8::StringObject::New(value).As<v8::StringObject>();
                           ^
/Users/akos.kitta/.node-gyp/10.13.0/include/node/v8.h:5052:3: note: 'New' has been explicitly marked deprecated here
  V8_DEPRECATED("Use Isolate* version",
  ^
/Users/akos.kitta/.node-gyp/10.13.0/include/node/v8config.h:327:29: note: expanded from macro 'V8_DEPRECATED'
  declarator __attribute__((deprecated))
                            ^
In file included from ../src/NSFW.cpp:1:
In file included from ../src/../includes/NSFW.h:5:
../../nan/nan.h:1066:44: warning: 'ToString' is deprecated [-Wdeprecated-declarations]
      v8::Local<v8::String> string = from->ToString();
                                           ^
/Users/akos.kitta/.node-gyp/10.13.0/include/node/v8.h:2454:10: note: 'ToString' has been explicitly marked deprecated here
  inline V8_DEPRECATED("Use maybe version", Local<String> ToString() const);
         ^
/Users/akos.kitta/.node-gyp/10.13.0/include/node/v8config.h:327:29: note: expanded from macro 'V8_DEPRECATED'
  declarator __attribute__((deprecated))
                            ^
In file included from ../src/NSFW.cpp:1:
In file included from ../src/../includes/NSFW.h:5:
../../nan/nan.h:1080:27: warning: 'WriteUtf8' is deprecated [-Wdeprecated-declarations]
        length_ = string->WriteUtf8(str_, static_cast<int>(len), 0, flags);
                          ^
/Users/akos.kitta/.node-gyp/10.13.0/include/node/v8.h:2658:3: note: 'WriteUtf8' has been explicitly marked deprecated here
  V8_DEPRECATED("Use Isolate* version",
  ^
/Users/akos.kitta/.node-gyp/10.13.0/include/node/v8config.h:327:29: note: expanded from macro 'V8_DEPRECATED'
  declarator __attribute__((deprecated))
                            ^
../src/NSFW.cpp:56:32: warning: 'Call' is deprecated [-Wdeprecated-declarations]
  baton->nsfw->mErrorCallback->Call(1, argv);
                               ^
../../nan/nan.h:1654:3: note: 'Call' has been explicitly marked deprecated here
  NAN_DEPRECATED inline v8::Local<v8::Value>
  ^
../../nan/nan.h:102:40: note: expanded from macro 'NAN_DEPRECATED'
# define NAN_DEPRECATED __attribute__((deprecated))
                                       ^
../src/NSFW.cpp:98:32: warning: 'Call' is deprecated [-Wdeprecated-declarations]
  baton->nsfw->mEventCallback->Call(1, argv);
                               ^
../../nan/nan.h:1654:3: note: 'Call' has been explicitly marked deprecated here
  NAN_DEPRECATED inline v8::Local<v8::Value>
  ^
../../nan/nan.h:102:40: note: expanded from macro 'NAN_DEPRECATED'
# define NAN_DEPRECATED __attribute__((deprecated))
                                       ^
../src/NSFW.cpp:180:34: warning: 'Uint32Value' is deprecated [-Wdeprecated-declarations]
  uint32_t debounceMS = info[0]->Uint32Value();
                                 ^
/Users/akos.kitta/.node-gyp/10.13.0/include/node/v8.h:2477:3: note: 'Uint32Value' has been explicitly marked deprecated here
  V8_DEPRECATED("Use maybe version", uint32_t Uint32Value() const);
  ^
/Users/akos.kitta/.node-gyp/10.13.0/include/node/v8config.h:327:29: note: expanded from macro 'V8_DEPRECATED'
  declarator __attribute__((deprecated))
                            ^
../src/NSFW.cpp:181:44: warning: 'ToString' is deprecated [-Wdeprecated-declarations]
  v8::String::Utf8Value utf8Value(info[1]->ToString());
                                           ^
/Users/akos.kitta/.node-gyp/10.13.0/include/node/v8.h:2454:10: note: 'ToString' has been explicitly marked deprecated here
  inline V8_DEPRECATED("Use maybe version", Local<String> ToString() const);
         ^
/Users/akos.kitta/.node-gyp/10.13.0/include/node/v8config.h:327:29: note: expanded from macro 'V8_DEPRECATED'
  declarator __attribute__((deprecated))
                            ^
../src/NSFW.cpp:181:25: warning: 'Utf8Value' is deprecated [-Wdeprecated-declarations]
  v8::String::Utf8Value utf8Value(info[1]->ToString());
                        ^
/Users/akos.kitta/.node-gyp/10.13.0/include/node/v8.h:2891:5: note: 'Utf8Value' has been explicitly marked deprecated here
    V8_DEPRECATED("Use Isolate version",
    ^
/Users/akos.kitta/.node-gyp/10.13.0/include/node/v8config.h:327:29: note: expanded from macro 'V8_DEPRECATED'
  declarator __attribute__((deprecated))
                            ^
../src/NSFW.cpp:212:15: warning: 'Call' is deprecated [-Wdeprecated-declarations]
    callback->Call(1, argv);
              ^
../../nan/nan.h:1654:3: note: 'Call' has been explicitly marked deprecated here
  NAN_DEPRECATED inline v8::Local<v8::Value>
  ^
../../nan/nan.h:102:40: note: expanded from macro 'NAN_DEPRECATED'
# define NAN_DEPRECATED __attribute__((deprecated))
                                       ^
../src/NSFW.cpp:258:15: warning: 'Call' is deprecated [-Wdeprecated-declarations]
    callback->Call(1, argv);
              ^
../../nan/nan.h:1654:3: note: 'Call' has been explicitly marked deprecated here
  NAN_DEPRECATED inline v8::Local<v8::Value>
  ^
../../nan/nan.h:102:40: note: expanded from macro 'NAN_DEPRECATED'
# define NAN_DEPRECATED __attribute__((deprecated))
                                       ^
../src/NSFW.cpp:260:15: warning: 'Call' is deprecated [-Wdeprecated-declarations]
    callback->Call(0, NULL);
              ^
../../nan/nan.h:1654:3: note: 'Call' has been explicitly marked deprecated here
  NAN_DEPRECATED inline v8::Local<v8::Value>
  ^
../../nan/nan.h:102:40: note: expanded from macro 'NAN_DEPRECATED'
# define NAN_DEPRECATED __attribute__((deprecated))
                                       ^
../src/NSFW.cpp:285:15: warning: 'Call' is deprecated [-Wdeprecated-declarations]
    callback->Call(1, argv);
              ^
../../nan/nan.h:1654:3: note: 'Call' has been explicitly marked deprecated here
  NAN_DEPRECATED inline v8::Local<v8::Value>
  ^
../../nan/nan.h:102:40: note: expanded from macro 'NAN_DEPRECATED'
# define NAN_DEPRECATED __attribute__((deprecated))
                                       ^
../src/NSFW.cpp:327:13: warning: 'Call' is deprecated [-Wdeprecated-declarations]
  callback->Call(0, NULL);
            ^
../../nan/nan.h:1654:3: note: 'Call' has been explicitly marked deprecated here
  NAN_DEPRECATED inline v8::Local<v8::Value>
  ^
../../nan/nan.h:102:40: note: expanded from macro 'NAN_DEPRECATED'
# define NAN_DEPRECATED __attribute__((deprecated))
                                       ^
13 warnings generated.
  CXX(target) Release/obj.target/nsfw/src/Queue.o
  CXX(target) Release/obj.target/nsfw/src/NativeInterface.o
  CXX(target) Release/obj.target/nsfw/src/osx/RunLoop.o
  CXX(target) Release/obj.target/nsfw/src/osx/FSEventsService.o
  SOLINK_MODULE(target) Release/nsfw.node
npm notice created a lockfile as package-lock.json. You should commit this file.
npm WARN nsfw-node-10@1.0.0 No description
npm WARN nsfw-node-10@1.0.0 No repository field.

+ nsfw@1.1.1
added 8 packages from 17 contributors and audited 9 packages in 9.867s
found 0 vulnerabilities

prprpr:nsfw-node-10 akos.kitta$ 
```

I will verify it within Theia.

No, it is failing because of `fiber`. I try to bump up the version. `wdio-mocha-framework` uses `fiber`.

Details:
```
prprpr:theia akos.kitta$ nvm use 10 && node --version && yarn --force
Now using node v10.13.0 (npm v6.4.1)
v10.13.0
yarn install v1.12.1
[1/5] 🔍  Validating package.json...
[2/5] 🔍  Resolving packages...
[3/5] 🚚  Fetching packages...
warning monaco-languageclient@0.9.0: The engine "vscode" appears to be invalid.
warning vscode-base-languageclient@4.4.0: The engine "vscode" appears to be invalid.
[4/5] 🔗  Linking dependencies...
warning " > istanbul-instrumenter-loader@3.0.1" has unmet peer dependency "webpack@^2.0.0 || ^3.0.0 || ^4.0.0".
warning " > tslint-language-service@0.9.9" has incorrect peer dependency "typescript@>= 2.3.1 < 3".
warning "workspace-aggregator-3aca36e3-bc66-4e74-b191-0a9f09c2d508 > @theia/application-manager > font-awesome-webpack@0.0.5-beta.2" has unmet peer dependency "font-awesome@>=4.3.0".
[5/5] 📃  Rebuilding all packages...
[1/13] ⡀ fibers
[2/13] ⡀ dtrace-provider
[6/13] ⡀ node-pty
[4/13] ⡀ electron
error /Users/akos.kitta/Desktop/theia/node_modules/fibers: Command failed.
Exit code: 127
Command: node build.js || nodejs build.js
Arguments:
Directory: /Users/akos.kitta/Desktop/theia/node_modules/fibers
Output:
gyp info it worked if it ends with ok
gyp info using node-gyp@3.8.0
gyp info using node@10.13.0 | darwin | x64
gyp WARN download NVM_NODEJS_ORG_MIRROR is deprecated and will be removed in node-gyp v4, please use NODEJS_ORG_MIRROR
gyp WARN download NVM_NODEJS_ORG_MIRROR is deprecated and will be removed in node-gyp v4, please use NODEJS_ORG_MIRROR
gyp info spawn /usr/bin/python
gyp info spawn args [ '/Users/akos.kitta/Desktop/theia/node_modules/node-gyp/gyp/gyp_main.py',
gyp info spawn args   'binding.gyp',
gyp info spawn args   '-f',
gyp info spawn args   'make',
gyp info spawn args   '-I',
gyp info spawn args   '/Users/akos.kitta/Desktop/theia/node_modules/fibers/build/config.gypi',
gyp info spawn args   '-I',
gyp info spawn args   '/Users/akos.kitta/Desktop/theia/node_modules/node-gyp/addon.gypi',
gyp info spawn args   '-I',
gyp info spawn args   '/Users/akos.kitta/.node-gyp/10.13.0/include/node/common.gypi',
gyp info spawn args   '-Dlibrary=shared_library',
gyp info spawn args   '-Dvisibility=default',
gyp info spawn args   '-Dnode_root_dir=/Users/akos.kitta/.node-gyp/10.13.0',
gyp info spawn args   '-Dnode_gyp_dir=/Users/akos.kitta/Desktop/theia/node_modules/node-gyp',
gyp info spawn args   '-Dnode_lib_file=/Users/akos.kitta/.node-gyp/10.13.0/<(target_arch)/node.lib',
gyp info spawn args   '-Dmodule_root_dir=/Users/akos.kitta/Desktop/theia/node_modules/fibers',
gyp info spawn args   '-Dnode_engine=v8',
gyp info spawn args   '--depth=.',
gyp info spawn args   '--no-parallel',
gyp info spawn args   '--generator-output',
gyp info spawn args   'build',
gyp info spawn args   '-Goutput_dir=.' ]
gyp WARN download NVM_NODEJS_ORG_MIRROR is deprecated and will be removed in node-gyp v4, please use NODEJS_ORG_MIRROR
gyp info spawn make
gyp info spawn args [ 'BUILDTYPE=Release', '-C', 'build' ]
  CXX(target) Release/obj.target/fibers/src/fibers.o
../src/fibers.cc:122:64: error: too few arguments to function call, expected atleast 3, have 2
                return String::NewFromOneByte(isolate, (const uint8_t*)string);
                       ~~~~~~~~~~~~~~~~~~~~~~                                ^
/Users/akos.kitta/.node-gyp/10.13.0/include/node/v8.h:2802:3: note: 'NewFromOneByte' declared here
  static V8_WARN_UNUSED_RESULT MaybeLocal<String> NewFromOneByte(
  ^
../src/fibers.cc:126:64: error: too few arguments to function call, expected atleast 3, have 2
                return String::NewFromOneByte(isolate, (const uint8_t*)string);
                       ~~~~~~~~~~~~~~~~~~~~~~                                ^
/Users/akos.kitta/.node-gyp/10.13.0/include/node/v8.h:2802:3: note: 'NewFromOneByte' declared here
  static V8_WARN_UNUSED_RESULT MaybeLocal<String> NewFromOneByte(
  ^
../src/fibers.cc:406:24: warning: 'Utf8Value' is deprecated [-Wdeprecated-declarations]
                                        String::Utf8Value stack(uni::Deref(that.isolate, fatal_stack));
                                                          ^
/Users/akos.kitta/.node-gyp/10.13.0/include/node/v8.h:2891:5: note: 'Utf8Value'has been explicitly marked deprecated here
    V8_DEPRECATED("Use Isolate version",
    ^
/Users/akos.kitta/.node-gyp/10.13.0/include/node/v8config.h:327:29: note: expanded from macro 'V8_DEPRECATED'
  declarator __attribute__((deprecated))
                            ^
../src/fibers.cc:433:80: error: no matching member function for call to 'NewInstance'
                                return uni::Return(uni::Deref(Isolate::GetCurrent(), tmpl)->GetFunction()->NewInstance(1, argv), args);
                                                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~
/Users/akos.kitta/.node-gyp/10.13.0/include/node/v8.h:3996:44: note: candidate function not viable: requires single argument 'context', but 2 arguments were provided
  V8_WARN_UNUSED_RESULT MaybeLocal<Object> NewInstance(
                                           ^
/Users/akos.kitta/.node-gyp/10.13.0/include/node/v8.h:3993:44: note: candidate function not viable: requires 3 arguments, but 2 were provided
  V8_WARN_UNUSED_RESULT MaybeLocal<Object> NewInstance(
                                           ^
../src/fibers.cc:621:14: error: no matching constructor for initialization of 'v8::TryCatch'
                                TryCatch try_catch;
                                         ^
/Users/akos.kitta/.node-gyp/10.13.0/include/node/v8.h:8536:12: note: candidate constructor not viable: requires single argument 'isolate', but no arguments were provided
  explicit TryCatch(Isolate* isolate);
           ^
/Users/akos.kitta/.node-gyp/10.13.0/include/node/v8.h:8656:3: note: candidate constructor not viable: requires 1 argument, but 0 were provided
  TryCatch(const TryCatch&) = delete;
  ^
../src/fibers.cc:628:13: warning: 'Compile' is deprecated [-Wdeprecated-declarations]
                                Script::Compile(uni::NewLatin1String(that.isolate, "void 0;"));
                                        ^
/Users/akos.kitta/.node-gyp/10.13.0/include/node/v8.h:1255:10: note: 'Compile' has been explicitly marked deprecated here
  static V8_DEPRECATED("Use maybe version",
         ^
/Users/akos.kitta/.node-gyp/10.13.0/include/node/v8config.h:327:29: note: expanded from macro 'V8_DEPRECATED'
  declarator __attribute__((deprecated))
                            ^
../src/fibers.cc:741:34: error: no matching member function for call to 'ToNumber'
                        Coroutine::pool_size = value->ToNumber()->Value();
                                               ~~~~~~~^~~~~~~~
/Users/akos.kitta/.node-gyp/10.13.0/include/node/v8.h:2425:44: note: candidate function not viable: requires single argument 'context', but no arguments were provided
  V8_WARN_UNUSED_RESULT MaybeLocal<Number> ToNumber(
                                           ^
/Users/akos.kitta/.node-gyp/10.13.0/include/node/v8.h:2442:31: note: candidate function not viable: requires single argument 'isolate', but no arguments were provided
                Local<Number> ToNumber(Isolate* isolate) const);
                              ^
../src/fibers.cc:793:72: error: too few arguments to function call, expected atleast 3, have 2
                        fn->SetAccessor(uni::NewLatin1Symbol(isolate, "current"), GetCurrent);
                        ~~~~~~~~~~~~~~~            ^
/Users/akos.kitta/.node-gyp/10.13.0/include/node/v8.h:3348:3: note: 'SetAccessor' declared here
  V8_WARN_UNUSED_RESULT Maybe<bool> SetAccessor(
  ^
/Users/akos.kitta/.node-gyp/10.13.0/include/node/v8config.h:416:31: note: expanded from macro 'V8_WARN_UNUSED_RESULT'
#define V8_WARN_UNUSED_RESULT __attribute__((warn_unused_result))
                              ^
../src/fibers.cc:794:63: error: no viable conversion from 'uni::FunctionType (Local<v8::String>, const uni::GetterCallbackInfo &)' (aka 'void (Local<v8::String>, const PropertyCallbackInfo<v8::Value> &)') to 'Local<v8::Name>'
                        fn->SetAccessor(uni::NewLatin1Symbol(isolate, "poolSize"), GetPoolSize, SetPoolSize);
   ^~~~~~~~~~~
/Users/akos.kitta/.node-gyp/10.13.0/include/node/v8.h:204:7: note: candidate constructor (the implicit copy constructor) not viable: no known conversion from 'uni::FunctionType (Local<v8::String>, const uni::GetterCallbackInfo &)' (aka 'void (Local<v8::String>, const PropertyCallbackInfo<v8::Value> &)') to 'const v8::Local<v8::Name> &' for 1st argument
class Local {
      ^
/Users/akos.kitta/.node-gyp/10.13.0/include/node/v8.h:204:7: note: candidate constructor (the implicit move constructor) not viable: no known conversion from 'uni::FunctionType (Local<v8::String>, const uni::GetterCallbackInfo &)' (aka 'void (Local<v8::String>, const PropertyCallbackInfo<v8::Value> &)') to 'v8::Local<v8::Name> &&' for 1st argument
/Users/akos.kitta/.node-gyp/10.13.0/include/node/v8.h:208:13: note: candidate template ignored: could not match 'Local<type-parameter-0-0>' against 'void (*)(v8::Local<v8::String>, const v8::PropertyCallbackInfo<v8::Value> &)'
  V8_INLINE Local(Local<S> that)
            ^
/Users/akos.kitta/.node-gyp/10.13.0/include/node/v8.h:3349:43: note: passing argument to parameter 'name' here
      Local<Context> context, Local<Name> name,
                                          ^
../src/fibers.cc:795:84: error: too few arguments to function call, expected atleast 3, have 2
                        fn->SetAccessor(uni::NewLatin1Symbol(isolate, "fibersCreated"), GetFibersCreated);
                        ~~~~~~~~~~~~~~~                        ^
/Users/akos.kitta/.node-gyp/10.13.0/include/node/v8.h:3348:3: note: 'SetAccessor' declared here
  V8_WARN_UNUSED_RESULT Maybe<bool> SetAccessor(
  ^
/Users/akos.kitta/.node-gyp/10.13.0/include/node/v8config.h:416:31: note: expanded from macro 'V8_WARN_UNUSED_RESULT'
#define V8_WARN_UNUSED_RESULT __attribute__((warn_unused_result))
                              ^
In file included from ../src/fibers.cc:1:
In file included from ../src/coroutine.h:1:
In file included from /Users/akos.kitta/.node-gyp/10.13.0/include/node/node.h:63:
/Users/akos.kitta/.node-gyp/10.13.0/include/node/v8.h:215:5: error: assigning to 'v8::Context *' from incompatible type 'v8::String *'
    TYPE_CHECK(T, S);
    ^~~~~~~~~~~~~~~~
/Users/akos.kitta/.node-gyp/10.13.0/include/node/v8.h:169:39: note: expanded from macro 'TYPE_CHECK'
    *(static_cast<T* volatile*>(0)) = static_cast<S*>(0);      \
                                      ^~~~~~~~~~~~~~~~~~
../src/fibers.cc:794:20: note: in instantiation of function template specialization 'v8::Local<v8::Context>::Local<v8::String>' requested here
                        fn->SetAccessor(uni::NewLatin1Symbol(isolate, "poolSize"), GetPoolSize, SetPoolSize);
                                        ^
2 warnings and 9 errors generated.
make: *** [Release/obj.target/fibers/src/fibers.o] Error 1
gyp ERR! build error
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/Users/akos.kitta/Desktop/theia/node_modules/node-gyp/lib/build.js:262:23)
gyp ERR! stack     at ChildProcess.emit (events.js:182:13)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:240:12)
gyp ERR! System Darwin 18.0.0
gyp ERR! command "/Users/akos.kitta/.nvm/versions/node/v10.13.0/bin/node" "/Users/akos.kitta/Desktop/theia/node_modules/.bin/node-gyp" "rebuild" "--release"
gyp ERR! cwd /Users/akos.kitta/Desktop/theia/node_modules/fibers
gyp ERR! node -v v10.13.0
gyp ERR! node-gyp -v v3.8.0
gyp ERR! not ok
node-gyp exited with code: 1
Please make sure you are using a supported platform and node version. If you
would like to compile fibers on this machine please make sure you have setup your
build environment--
Windows + OS X instructions here: https://github.com/nodejs/node-gyp
Ubuntu users please run: `sudo apt-get install g++ build-essential`



prprpr:theia akos.kitta$ yarn why fibers
yarn why v1.12.1
[1/4] 🤔  Why do we have the module "fibers"...?
[2/4] 🚚  Initialising dependency graph...
[3/4] 🔍  Finding dependency...
[4/4] 🚡  Calculating file sizes...
=> Found "fibers@1.0.15"
info Reasons this module exists
   - "_project_#wdio-mocha-framework#wdio-sync" depends on it
   - Hoisted from "_project_#wdio-mocha-framework#wdio-sync#fibers"
info Disk size without dependencies: "1.5MB"
info Disk size with unique dependencies: "1.5MB"
info Disk size with transitive dependencies: "1.5MB"
info Number of shared dependencies: 0
✨  Done in 2.11s.
prprpr:theia akos.kitta$
```

--- a/.gitpod.yml
+++ b/.gitpod.yml
@@ -1,8 +1,8 @@
 ports:
 - port: 3000
 tasks:
-- init: yarn install
-  command: cd examples/browser && yarn start --hostname 0.0.0.0 ../..
+- init: nvm install 10 && nvm use 10 && yarn
+  command: yarn --cwd examples/browser start ../..
 github:
   prebuilds:
     pullRequestsFromForks: true

--- a/.travis.yml
+++ b/.travis.yml
@@ -1,6 +1,6 @@
 sudo: required
 language: node_js
-node_js: '8'
+node_js: '10'
 git:
   depth: 1
 cache:

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -4,12 +4,12 @@
 
 - [filesystem] added the menu item `Upload Files...` to easily upload files into a workspace
 - [workspace] allow creation of files and folders using recursive paths
+- [electron] upgraded version of electron used to version 3.
 
 Breaking changes:
 
+- [node] moved to using Node.js version 10, dropping support for Node.js version 8.
 - [dialog] `validate` and `accept` methods are now Promisified [#4764](https://github.com/theia-ide/theia/pull/4764)
-
-Breaking changes:
 - [editor] turn off autoSave by default to align with VS Code [#4777](https://github.com/theia-ide/theia/pull/4777)
   - default settings can be overriden in application package.json:
   ```json

--- a/appveyor.yml
+++ b/appveyor.yml
@@ -2,7 +2,7 @@ shallow_clone: true
 
 environment:
   matrix:
-    - nodejs_version: "8"
+    - nodejs_version: "10"
 
 platform:
   - x64

--- a/dev-packages/application-manager/package.json
+++ b/dev-packages/application-manager/package.json
@@ -35,7 +35,7 @@
     "circular-dependency-plugin": "^5.0.0",
     "copy-webpack-plugin": "^4.5.0",
     "css-loader": "^0.28.1",
-    "electron": "^2.0.14",
+    "electron": "^3.1.7",
     "electron-rebuild": "^1.5.11",
     "file-loader": "^1.1.11",
     "font-awesome-webpack": "0.0.5-beta.2",

--- a/doc/Developing.md
+++ b/doc/Developing.md
@@ -44,10 +44,9 @@ For Windows instructions [click here](#building-on-windows).
      - [Root privileges errors](#root-privileges-errors)
 
 ## Prerequisites
- - Node.js `>= 8.x`, `< 9.x`.
-   - Preferably, **use** version `8.11.4`, it has the [active LTS](https://github.com/nodejs/Release).
-   - Node.js `9.x` is untested.
-   - Node.js `10.x` is **not** supported yet due to a known issue in [`nsfw`](https://github.com/theia-ide/theia/issues/2009).
+ - Node.js `>= 10.2.0`.
+   - Preferably, **use** version `10.15.3`, it has the [active LTS](https://github.com/nodejs/Release).
+   - Node.js `11.x` is untested.
  - [Yarn package manager](https://yarnpkg.com/en/docs/install) v1.7.0
  - git (If you would like to use the Git-extension too, you will need to have git version 2.11.0 or higher.)
 

--- a/package.json
+++ b/package.json
@@ -4,17 +4,17 @@
   "version": "0.0.0",
   "engines": {
     "yarn": "1.0.x || >=1.2.1",
-    "node": ">=8.9.3"
+    "node": ">=10.2.0"
   },
   "resolution": {
-    "**/@types/node": "8.10.20"
+    "**/@types/node": "~10.3.6"
   },
   "devDependencies": {
     "@types/chai": "^4.0.1",
     "@types/chai-string": "^1.4.0",
     "@types/jsdom": "^11.0.4",
     "@types/mocha": "^2.2.41",
-    "@types/node": "8.10.20",
+    "@types/node": "~10.3.6",
     "@types/sinon": "^2.3.5",
     "@types/temp": "^0.8.29",
     "@types/webdriverio": "^4.7.0",
@@ -39,7 +39,7 @@
     "typedoc-plugin-external-module-map": "^1.0.0",
     "typescript": "^3.1.3",
     "uuid": "^3.1.0",
-    "wdio-mocha-framework": "0.5.13",
+    "wdio-mocha-framework": "0.6.4",
     "wdio-selenium-standalone-service": "0.0.12",
     "wdio-spec-reporter": "0.1.5",
     "webdriverio": "4.14.1"

--- a/packages/core/package.json
+++ b/packages/core/package.json
@@ -20,7 +20,7 @@
     "@types/yargs": "^11.1.0",
     "ajv": "^6.5.3",
     "body-parser": "^1.17.2",
-    "electron": "^2.0.14",
+    "electron": "^3.1.7",
     "electron-store": "^2.0.0",
     "es6-promise": "^4.2.4",
     "express": "^4.16.3",

--- a/packages/core/src/browser/tree/tree.spec.ts
+++ b/packages/core/src/browser/tree/tree.spec.ts
@@ -174,8 +174,7 @@ describe('Tree', () => {
       const expectedRefreshedNodes = new Set([
         retrieveNode<CompositeTreeNode>('1'),
         retrieveNode<CompositeTreeNode>('1.1'),
-        retrieveNode<CompositeTreeNode>('1.2'),
-        retrieveNode<CompositeTreeNode>('1.2.1')]);
+        retrieveNode<CompositeTreeNode>('1.2')]);
       model.onNodeRefreshed((e: Readonly<CompositeTreeNode>) => {
         result = result && expectedRefreshedNodes.has(e);
         expectedRefreshedNodes.delete(e);

--- a/packages/core/src/node/backend-application.ts
+++ b/packages/core/src/node/backend-application.ts
@@ -24,6 +24,7 @@ import { ILogger, ContributionProvider, MaybePromise } from '../common';
 import { CliContribution } from './cli';
 import { Deferred } from '../common/promise-util';
 import { environment } from '../common/index';
+import { AddressInfo } from 'net';
 
 export const BackendApplicationContribution = Symbol('BackendApplicationContribution');
 export interface BackendApplicationContribution {
@@ -185,7 +186,7 @@ export class BackendApplication {
 
         server.listen(port, hostname, () => {
             const scheme = this.cliParams.ssl ? 'https' : 'http';
-            this.logger.info(`Theia app listening on ${scheme}://${hostname || 'localhost'}:${server.address().port}.`);
+            this.logger.info(`Theia app listening on ${scheme}://${hostname || 'localhost'}:${(server.address() as AddressInfo).port}.`);
             deferred.resolve(server);
         });
 

--- a/packages/core/src/node/messaging/test/test-web-socket-channel.ts
+++ b/packages/core/src/node/messaging/test/test-web-socket-channel.ts
@@ -19,6 +19,7 @@ import * as http from 'http';
 import * as https from 'https';
 import { WebSocketChannel } from '../../../common/messaging/web-socket-channel';
 import { Disposable } from '../../../common/disposable';
+import { AddressInfo } from 'net';
 
 export class TestWebSocketChannel extends WebSocketChannel {
 
@@ -27,7 +28,7 @@ export class TestWebSocketChannel extends WebSocketChannel {
         path: string
     }) {
         super(0, content => socket.send(content));
-        const socket = new ws(`ws://localhost:${server.address().port}${WebSocketChannel.wsPath}`);
+        const socket = new ws(`ws://localhost:${(server.address() as AddressInfo).port}${WebSocketChannel.wsPath}`);
         socket.on('error', error =>
             this.fireError(error)
         );

--- a/packages/git/src/electron-node/askpass/askpass.ts
+++ b/packages/git/src/electron-node/askpass/askpass.ts
@@ -14,6 +14,7 @@ import { MaybePromise } from '@theia/core/lib/common/types';
 import { Deferred } from '@theia/core/lib/common/promise-util';
 import { GitPrompt } from '../../common/git-prompt';
 import { DugiteGitPromptServer } from '../../node/dugite-git-prompt';
+import { AddressInfo } from 'net';
 
 /**
  * Environment for the Git askpass helper.
@@ -89,7 +90,7 @@ export class Askpass implements Disposable {
             return new Promise<Address>(resolve => {
                 this.server.on('error', err => this.logger.error(err));
                 this.server.listen(0, this.hostname(), () => {
-                    resolve(this.server.address());
+                    resolve(this.server.address() as AddressInfo);
                 });
             });
         } catch (err) {

--- a/packages/java/src/node/java-contribution.ts
+++ b/packages/java/src/node/java-contribution.ts
@@ -28,6 +28,7 @@ import { JAVA_LANGUAGE_ID, JAVA_LANGUAGE_NAME, JavaStartParams } from '../common
 import { JavaCliContribution } from './java-cli-contribution';
 import { ContributionProvider } from '@theia/core';
 import { JavaExtensionContribution } from './java-extension-model';
+import { AddressInfo } from 'net';
 const sha1 = require('sha1');
 
 export type ConfigurationType = 'config_win' | 'config_mac' | 'config_linux';
@@ -137,8 +138,8 @@ export class JavaContribution extends BaseLanguageServerContribution {
         this.logInfo('logs at ' + path.resolve(workspacePath, '.metadata', '.log'));
         const env = Object.create(process.env);
         const address = server.address();
-        env.CLIENT_HOST = address.address;
-        env.CLIENT_PORT = address.port;
+        env.CLIENT_HOST = (address as AddressInfo).address;
+        env.CLIENT_PORT = (address as AddressInfo).port;
         const serverConnection = await this.createProcessSocketConnection(socket, socket, command, args, { env });
         this.forward(clientConnection, serverConnection);
     }

--- a/packages/preview/src/browser/markdown/markdown-preview-handler.ts
+++ b/packages/preview/src/browser/markdown/markdown-preview-handler.ts
@@ -106,7 +106,7 @@ export class MarkdownPreviewHandler implements PreviewHandler {
         if (!elementToReveal) {
             return;
         }
-        elementToReveal.scrollIntoView({ behavior: 'instant' });
+        elementToReveal.scrollIntoView();
     }
 
     findElementForFragment(content: HTMLElement, link: string): HTMLElement | undefined {

--- a/packages/preview/src/browser/preview-widget.ts
+++ b/packages/preview/src/browser/preview-widget.ts
@@ -208,7 +208,7 @@ export class PreviewWidget extends BaseWidget implements Navigatable {
         const elementToReveal = this.previewHandler.findElementForFragment(this.node, uri.fragment);
         if (elementToReveal) {
             this.preventScrollNotification = true;
-            elementToReveal.scrollIntoView({ behavior: 'instant' });
+            elementToReveal.scrollIntoView();
             window.setTimeout(() => {
                 this.preventScrollNotification = false;
             }, 50);
@@ -225,7 +225,7 @@ export class PreviewWidget extends BaseWidget implements Navigatable {
         const elementToReveal = this.previewHandler.findElementForSourceLine(this.node, sourceLine);
         if (elementToReveal) {
             this.preventScrollNotification = true;
-            elementToReveal.scrollIntoView({ behavior: 'instant' });
+            elementToReveal.scrollIntoView();
             window.setTimeout(() => {
                 this.preventScrollNotification = false;
             }, 50);

--- a/packages/process/src/node/raw-process.ts
+++ b/packages/process/src/node/raw-process.ts
@@ -119,9 +119,9 @@ export class RawProcess extends Process {
                 );
             });
 
-            this.output = this.process.stdout;
-            this.input = this.process.stdin;
-            this.errorOutput = this.process.stderr;
+            this.output = this.process.stdout || new DevNullStream();
+            this.input = this.process.stdin || new DevNullStream();
+            this.errorOutput = this.process.stderr || new DevNullStream();
 
             if (this.process.pid !== undefined) {
                 process.nextTick(() => {

